sound/soc/sof/topology.c

```diff
@@ -2445,6 +2437,7 @@ static int sof_widget_unload(struct snd_soc_component *scomp,
 		dai = swidget->private;
 
 		if (dai) {
+			kfree(dai->comp_dai);
```

Comment:
maybe I missed it but I don't see a matching kzalloc for dai->comp_dai?

Reply:
yes, it is here: https://github.com/thesofproject/linux/blob/95e1be1602f61693aa361e168efed63554119fd2/sound/soc/sof/topology.c#L1484